remote-wallet/src/remote_wallet.rs

```diff
@@ -104,126 +55,239 @@ impl RemoteWalletManager {
-            .map(|d| d.info.clone())
+            .find(|device| &device.info.pubkey == pubkey)
+            .map(|device| device.info.clone())
     }
 
-    /// Update devices in maximum `max_polling_duration` if it doesn't succeed
+    /// Get the total number of connected devices
+    pub fn device_count(&self) -> usize {
+        self.devices.read().len()
+    }
+
+    /// Check if any devices are connected
+    pub fn has_devices(&self) -> bool {
+        !self.devices.read().is_empty()
+    }
+
+    /// Get devices by manufacturer
+    pub fn get_devices_by_manufacturer(
+        &self,
+        manufacturer: &Manufacturer,
+    ) -> Vec<RemoteWalletInfo> {
+        self.devices
+            .read()
+            .iter()
+            .filter(|device| &device.info.manufacturer == manufacturer)
+            .map(|device| device.info.clone())
+            .collect()
+    }
+
+    /// Get the first available wallet of any type
+    ///
+    /// This is a convenience method that returns the first connected wallet,
+    /// regardless of its type. Useful when you just need any available wallet.
+    pub fn get_first_available_wallet(&self) -> Option<RemoteWalletInfo> {
+        self.devices
+            .read()
+            .first()
+            .map(|device| device.info.clone())
+    }
+
+    /// Attempt to connect to hardware wallets with polling within a time limit
+    ///
+    /// This method will continuously attempt to discover and connect to hardware wallets
+    /// until either devices are found or the maximum polling duration is exceeded.
+    ///
+    /// Returns `true` if at least one device was successfully connected, `false` otherwise
     pub fn try_connect_polling(&self, max_polling_duration: &Duration) -> bool {
         let start_time = Instant::now();
         while start_time.elapsed() <= *max_polling_duration {
-            if let Ok(num_devices) = self.update_devices() {
-                let plural = if num_devices == 1 { "" } else { "s" };
-                trace!("{num_devices} Remote Wallet{plural} found");
-                return true;
+            match self.update_devices() {
+                Ok(_new_device_count) => {
+                    let current_total = self.devices.read().len();
+                    if current_total > 0 {
+                        let plural = if current_total == 1 {
+                            LOG_DEVICE_SINGULAR
+                        } else {
+                            LOG_DEVICE_PLURAL
+                        };
+                        trace!("{current_total} Remote Wallet{plural} found");
+                        return true;
+                    }
+                }
+                Err(err) => {
+                    debug!("Error during device discovery: {err:?}");
+                    // Continue trying despite errors
+                }
             }
+
+            // Small delay to avoid excessive polling
+            std::thread::sleep(Duration::from_millis(100));
```

Comment:
If we want to back off, we should let the sleep be configurable so that people's code isn't stopping in confusing ways. For example, remote-wallet is used in async runtimes in the CLI, so doing a thread::sleep could stop the whole async runtime.

Unless there's a particular use-case you're worried about, let's remove this